vec3d.h

```diff
@@ -0,0 +1,130 @@
+#include<math.h>
+
+
+template<typename T>
+class Vec3{
+    private:
+        T x, y, z;
+    public:
+        Vec3() : x(T(0)), y(T(0)), z(T(0)) {}           
+        Vec3(T xx) : x(xx), y(xx), z(xx) {}
+        Vec3(T xx, T yy, T zz): x(xx), y(yy), z(zz) {}
+
+        T getx(){return x;}
+        T gety(){return y;}
+        T getz(){return z;}
+        void xin(T &xx){x = xx;}
+        void yin(T &yy){y = yy;}
+        void zin(T &zz){z = zz;}
+
+
+
+
+        Vec3<T> operator * (const T &f){
```

Comment:
I'm pretty sure the `<T>` isn't necessary here because the compiler can deduce the template type.